example-mlem-get-started/generate.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+# See https://dvc.org/doc/tutorials/get-started
+
+# Setup script env:
+#   e   Exit immediately if a command exits with a non-zero exit status.
+#   u   Treat unset variables as an error when substituting.
+#   x   Print commands and their arguments as they are executed.
+set -eux
+
+HERE="$( cd "$(dirname "$0")" ; pwd -P )"
+REPO_NAME="example-mlem-get-started"
+
+BUILD_PATH="$HERE/build"
+
+pushd $BUILD_PATH
+if [ ! -d "$BUILD_PATH/.venv" ]; then
+  exit 1
+  virtualenv -p python3 .venv
+  export VIRTUAL_ENV_DISABLE_PROMPT=true
+  source .venv/bin/activate
+  echo '.venv/' > .gitignore
+  pip install gitpython
+  pip install "git+https://github.com/iterative/mlem#egg=mlem[all]" --use-deprecated=legacy-resolver
+  pip install -r $HERE/code/src/requirements.txt
+fi
+popd
+
+source $BUILD_PATH/.venv/bin/activate
+
+REPO_PATH="$HERE/build/$REPO_NAME"
+
+if [ -d "$REPO_PATH" ]; then
+  echo "Repo $REPO_PATH already exists, please remove it first."
+  exit 1
+fi
+
+TOTAL_TAGS=15
+STEP_TIME=100000
+BEGIN_TIME=$(( $(date +%s) - ( ${TOTAL_TAGS} * ${STEP_TIME}) ))
+export TAG_TIME=${BEGIN_TIME}
+export GIT_AUTHOR_DATE=${TAG_TIME}
+export GIT_COMMITTER_DATE=${TAG_TIME}
+tick(){
+  export TAG_TIME=$(( ${TAG_TIME} + ${STEP_TIME} ))
+  export GIT_AUTHOR_DATE=${TAG_TIME}
+  export GIT_COMMITTER_DATE=${TAG_TIME}
+}
+
+export GIT_AUTHOR_NAME="Mikhail Sveshnikov"
+export GIT_AUTHOR_EMAIL="mike0sv@gmail.com"
+export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"
+export GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"
+
+
+
+mkdir -p $REPO_PATH
+pushd $REPO_PATH
+
+git init -b main
+cp $HERE/code/README.md .
+cp $HERE/.gitattributes .
+git add .
+tick
+git commit -m "Initialize Git repository"
+git tag -a "0-git-init" -m "Git initialized."
+
+mlem init
+tick
+git add .mlem/config.yaml
+git commit -m "Initialize MLEM project"
+git tag -a "1-mlem-init" -m "MLEM initialized."
+
+
+cp $HERE/code/src/requirements.txt .
+cp $HERE/code/src/prepare.py .
```

Comment:
If you don it similar to DVC get-started you need to bundle this code and share from S3 so that people could curl or wget it

Reply:
raw.github.com works as well, I don't see any particular problems with that .. I would bundle though everything as a single tar (or may github has a way to download revisions as a tar with curl, I don't know)

Reply:
https://github.com/iterative/example-mlem-get-started/archive/refs/tags/5-switch-to-dvc.tar.gz for example

Reply:
it contains some extra files that are not needed initially though?

Reply:
it contains whole repo, yes. But I don't see any point of maintaining them separately, since they are very small and easy and also will be available in the docs themselves

Reply:
it's not about the size, it's about the workflow. In the dvc.org get started you would have a command:

```
wget code.zip
unzip code.zip
git add -a -m "add code"
```

https://dvc.org/doc/start/data-pipelines#expand-to-download-example-code

including any extra files there would confuse and ruin the flow of the document. 

so, your call here - if you want something similar to DVC then consider creating a clean code package.

Reply:
thinking second time - I remember now why I didn't use GH (most likely) - I wanted to keep the command as short and nice as possible `wget https://code.dvc.org/get-started/code.zip` and zip instead of tar